nakadi/event_stream.py

```diff
@@ -80,47 +77,31 @@ def generator():
 
                     # init new batch
                     messages_read_in_batch = 0
-                    current_batch = {}
-                    for partition in partitions:
-                        current_batch[partition] = []
-                    batch_start_time = datetime.datetime.now()
+                    current_batch = {partition: [] for partition in partitions}
+                    batch_start_time = time.time()
 
                     yield BATCH_SEPARATOR
 
                 # check if we reached the stream timeout or message count limit
-                time_since_start = datetime.datetime.now() - start
-                if time_since_start.total_seconds() >= opts['stream_timeout'] > 0 or 0 < opts['stream_limit'] <= messages_read:
+                time_since_start = time.time() - start
+                if time_since_start >= opts['stream_timeout'] > 0 or 0 < opts['stream_limit'] <= messages_read:
 
                     if messages_read_in_batch > 0:
-                        for partition in partitions:
-                            topic_partition = (topic.encode('UTF-8'), partition)
-                            # send the messages we could read so far
-                            if len(current_batch[partition]) > 0:
-
-                                stream_message = __create_stream_message(partition, latest_offsets[topic_partition], current_batch[partition])
-                                with __measure_time(current_batch[partitions], stream_message):
-                                    yield stream_message
-
-                            # just send the keep alive
-                            else:
-                                yield __create_stream_message(partition, latest_offsets[topic_partition])
+                        yield from process_batch(latest_offsets, current_batch)
+
                     break
 
     return generator()
 
 
-def __create_stream_message(partition, offset, events = None, topology = None):
-    message = {
-        'cursor': {
-            'partition': str(partition),
-            'offset': str(offset)
-        }
-    }
+def __create_stream_message(partition, offset, events=None, topology=None):
+    message = ['{"cursor":{"partition":"', str(partition), '","offset":"', str(offset), '"}']
```

Comment:
Is there a high performance method to escape a string safely as a JSON string when marshalling offsets?